### PR TITLE
Add feature to filter videos by date

### LINKED ISSUE #31: Sub select videos by date
I would like to be able to run youtube2zim and only gather newer that a certain date.
This would allow splitting a youtube channel zim into smaller parts. Also it would allow creating a new zim for newer content only, without having to modify already downloaded youtube channel zims.

### PR DESCRIPTION
**Pull request by deborahlow97**
Fixes #31 

### Context
- Allow user to add a parameter to filter and download videos uploaded on or after the specified date.

### Approach
- Used `youtube_dl.DateRange` to specify daterange
- Used Youtube API to get published date

### CODE REVIEW

**Comment by rgaudin** [comment]
Great! It's working fine. Couple things I'd love though:

* `DateRange` is very permissive but can still fail with garbage value. Could you protect that and make sure user gets an informative message in case of incorrect value? I'd prefer that to happen in `run()` so you could just store it in `__init__` and convert/fail to `DateRange` in `run`
* Logs should display the daterange as well ; in the same way we display the nb of playlists/videos.
* in `filter_videos_not_within_date_range`, I see no need for your `date_range` variable. Can we get rid of it? Actually, I think you used it for sementic purposes and that makes me think we should maybe store the date range in a `self.date_range` instead. The parameter can stay as it is, as it implies youtube-dl compatibility.

**Comment by deborahlow97** [comment]
> Great! It's working fine. Couple things I'd love though:
> 
> * `DateRange` is very permissive but can still fail with garbage value. Could you protect that and make sure user gets an informative message in case of incorrect value? I'd prefer that to happen in `run()` so you could just store it in `__init__` and convert/fail to `DateRange` in `run`
> * Logs should display the daterange as well ; in the same way we display the nb of playlists/videos.
> * in `filter_videos_not_within_date_range`, I see no need for your `date_range` variable. Can we get rid of it? Actually, I think you used it for sementic purposes and that makes me think we should maybe store the date range in a `self.date_range` instead. The parameter can stay as it is, as it implies youtube-dl compatibility.

👍 Alright, will do! Thanks for the informative feedback 😄

File: youtube2zim/scraper.py [hunk 1]
```diff
@@ -387,7 +400,7 @@ def extract_playlists(self):
         """ prepare a list of Playlist from user request
 
             USER: we fetch the hidden channel associate to it
-            CHANNEL (and USER): we grab all playlists + `uploads` playlist
+            CHANNEL (and USER): we grab all playlistFs + `uploads` playlist
```
Comment (rgaudin) [inline youtube2zim/scraper.py]: typo here?

**Comment by deborahlow97** [comment]
Currently, if no `--dateafter` is provided, `INFO:videos in date range: 0001-01-01 - 9999-12-31` would be logged.

**Comment by rgaudin** [comment]
Thank you @deborahlow97 !

**Pull request merged**